packages/react/src/components/BarChartCard/barChartUtils.js

```diff
@@ -241,7 +241,10 @@ export const formatChartData = (
         group.forEach((value) => {
           series.forEach((dataset) => {
             // if value is null, don't add it to the formatted chartData
-            if (!isNil(value[dataset.dataSourceId])) {
+            if (
+              !isNil(value[dataset.dataSourceId]) &&
+              (!isNil(value[categoryDataSourceId]) || timeDataSourceId)
```

Comment:
may be ? 

```
(!isNil(value[categoryDataSourceId]) || !isNil(value[timeDataSourceId]))
```

Reply:
What purpose does this second `!isNil(value[categoryDataSourceId])` in the ternary serve? This change is making this if statement look pretty complicated. Can we extract this out into a variable named in a descriptive way and use that in the if statement? ie. 
``` js
const isNilNonGroupedTimeDataSourceIdValue = timeDataSourceId && type !== BAR_CHART_TYPES.GROUPED ? !isNil(value[timeDataSourceId]) : true;
```

Reply:
Just as an FYI, this ternary is creating a new "branch", that if not checked by some test will lower the code coverage of this file. These small changes with no test are what can lead to coverage issues that are seemingly unrelated down the road.

Reply:
> What purpose does this second `!isNil(value[categoryDataSourceId])` in the ternary serve? This change is making this if statement look pretty complicated. Can we extract this out into a variable named in a descriptive way and use that in the if statement? ie.
> 
> ```js
> const nilCheckNonGroupedTimeDataSourceIdValue = timeDataSourceId && type !== BAR_CHART_TYPES.GROUPED ? !isNil(value[timeDataSourceId]) : true; // false if we want the check to fail without this
> ```

For Non-timeseries: if the data does not have value[categoryDataSourceId], it is assigning undefined and does not loads grouped bar chart at all. So `!isNil(value[categoryDataSourceId])` checks before assigning.
Also i will assign it to a variable as you suggested and add test cases too.